### PR TITLE
Add metrics plotter

### PR DESCRIPTION
**Pull request by evgenabramov**
Before submitting a pull request, please do the following steps:

1. Read [instructions for contributors](https://catboost.ai/docs/concepts/development-and-contributions.html).
2. Run `ya make` in catboost folder to make sure the code builds.
3. Add tests that test your change.
4. Run tests using `ya make -t -A` command.
5. If you haven't already, complete the [CLA](https://yandex.ru/legal/cla/).

### CODE REVIEW

File: catboost/python-package/catboost/widget/callbacks.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
+import xgboost as xgb
```
Comment (kizill) [inline catboost/python-package/catboost/widget/callbacks.py:1]: Please make this import optional to make catboost widget independent of xgboost installation.

File: catboost/python-package/catboost/widget/metrics_plotter.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+
+import typing as tp
+
+from traitlets import Unicode, Dict, default
+from IPython.display import display
+from ipywidgets import DOMWidget, Layout, widget_serialization
+from copy import deepcopy
+from typing import List, Optional, Any, Union
+
+logging.getLogger().setLevel(logging.DEBUG)
```
Comment (kizill) [inline catboost/python-package/catboost/widget/metrics_plotter.py:12]: Not needed, i guess.

File: catboost/python-package/catboost/widget/metrics_plotter.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+class MetricsWidget(DOMWidget):
```
Comment (kizill) [inline catboost/python-package/catboost/widget/metrics_plotter.py:14]: Maybe use common base class for `MetricsWidget` and `MetricVisualizer` to store repeated defs?

File: catboost/python-package/catboost/widget/metrics_plotter.py (hunk 2 above)
Comment (kizill) [inline catboost/python-package/catboost/widget/metrics_plotter.py:12]: Maybe create special logger instance for plotter:
```logger = logging.getLogger(__name__)```

**Review by kizill — CHANGES_REQUESTED** [review]

**Comment by kizill** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@evgenabramov, please agree to the CLA (https://github.com/catboost/catboost/blob/master/CONTRIBUTING.md#provide-contributions)

**Comment by evgenabramov** [comment]
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=ru.

**Review by kizill — APPROVED** [review]

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@kizill, internal review request updated: [2427817](https://a.yandex-team.ru/review/2427817)

**Pull request merged**